app/src/main/java/ch/epfl/sweng/erpa/activities/GameListActivity.java

```diff
@@ -34,25 +37,30 @@
     @BindView(R.id.recyclerView) RecyclerView mRecyclerView;
     private List<Game> games;
     private Resources resources;
+    private Toolbar toolbar;
+    private GameList gameList;
+    private Bundle bundle;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         if (dependenciesNotReady()) return;
         DataBindingUtil.setContentView(this, R.layout.activity_game_list);
         ButterKnife.bind(this);
-        onResume();
     }
 
     @Override
     protected void onResume() {
         super.onResume();
         if (dependenciesNotReady()) return;
         Intent myIntent = getIntent();
-        Bundle bundle = myIntent.getExtras();
+        bundle = myIntent.getExtras();
 
         if (bundle != null) {
-            GameList gameList = (GameList) bundle.getSerializable(GAME_LIST_ACTIVTIY_CLASS_KEY);
+            gameList = (GameList) bundle.getSerializable(GAME_LIST_ACTIVTIY_CLASS_KEY);
+            toolbar = findViewById(R.id.game_list_toolbar);
+            setSupportActionBar(toolbar);
+            getSupportActionBar().setDisplayShowTitleEnabled(false);
             setToolbarText(gameList);
```

Comment:
What happend if bundle is null? Do you crash? Do you refuse to launch? Please add a behaviour if the activity is launch without an extra

Reply:
After offline discussion, this is not cause any damage to the representation invariant. More test are needed though